### PR TITLE
[FrontPanel] Added FrontPanel plugin with HAL.

### PR DESCRIPTION
**Pull request by aswaunni**
Front Panel Control plugin using RDK HAL.
Implemented and tested the plugin in buildroot using dummy library (libds-hal.so) with api implementations according to dsFPConfig file and using the necessary headers (dsError.h, dsTypes.h, dsFPD.h ). Since its not part of the plugin, I have attached these files as zip.
 
[Files.zip](https://github.com/WebPlatformForEmbedded/WPEFrameworkPlugins/files/2256030/Files.zip)

### CODE REVIEW

File: FrontPanel/FrontPanel.cpp [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    Core::ProxyType<Web::Response> FrontPanel::PostMethod(Core::TextSegmentIterator& index, const Web::Request& request)
+    {
+        Core::ProxyType<Web::Response> result(PluginHost::Factories::Instance().Response());
+        result->ErrorCode = Web::STATUS_BAD_REQUEST;
+        result->Message = _T("Unsupported POST requestservice.");
+
+        if (index.IsValid() == true) {
+            if (index.Next() && index.IsValid()) {
+                Core::ProxyType<Web::JSONBodyType<FrontPanel::Config>> response(jsonBodyDataFactory.Element());
+                if (index.Remainder() == _T("GetFPState") && (request.HasBody())) {
```
Comment (pwielders) [inline FrontPanel/FrontPanel.cpp]: Same as previous remark, suggest to cal this "State", can be used, using a GET to retrieve and POST for the set.

File: FrontPanel/FrontPanel.cpp [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+                    if (_controller->GetFPState(FPDIndicator, state)) {
+                        response->FPDState = state;
+                        result->ErrorCode = Web::STATUS_OK;
+                        result->Body(response);
+                        result->Message = _T("Get Front Panel State");
+                    } else {
+                        result->ErrorCode = Web::STATUS_NO_CONTENT;
+                        result->Message = _T("Failed to Get Front Panel State.");
+                    }
+                } else if (index.Remainder() == _T("GetFPBrightness") && (request.HasBody())) {
```
Comment (pwielders) [inline FrontPanel/FrontPanel.cpp]: Applicable to all methods on the RESTFull interface, suggest removing Get/Set (indicated by the HTTP verb) and FB (indicated by the Callsign)

File: FrontPanel/FrontPanel.cpp (hunk 1 above)
Comment (aswaunni) [reply FrontPanel/FrontPanel.cpp]: Sure.

Comment (aswaunni) [reply FrontPanel/FrontPanel.cpp]: I have made the changes and committed.

File: FrontPanel/FrontPanel.cpp [hunk 3]
```diff
@@ -67,202 +67,216 @@ namespace Plugin {
...
                     } else {
                         result->ErrorCode = Web::STATUS_NO_CONTENT;
                         result->Message = _T("Failed to Set Front Panel Scroll.");
                     }
-                } else if (index.Remainder() == _T("SetFPDBrightness") && (request.HasBody())) {
-                    uint32_t FPDIndicator = request.Body<const Config>()->FPDIndicator.Value();
-                    uint32_t FPDBrightness = request.Body<const Config>()->FPDBrightness.Value();
-                    bool ToPersist = request.Body<const Config>()->ToPersist.Value();
-                    if (_controller->SetFPDBrightness(FPDIndicator, FPDBrightness, ToPersist)) {
+                } else if (index.Current() == _T("FPDBrightness") && (request.HasBody())) {
```
Comment (pwielders) [inline FrontPanel/FrontPanel.cpp]: Make this method DBrightness. FP is already in the callsign

File: FrontPanel/FrontPanel.cpp [hunk 4]
```diff
@@ -67,202 +67,216 @@ namespace Plugin {
...
                     } else {
                         result->ErrorCode = Web::STATUS_NO_CONTENT;
                         result->Message = _T("Failed to Set Front Panel Brightness.");
                     }
-                } else if (index.Remainder() == _T("SetFPDColor") && (request.HasBody())) {
-                    uint32_t FPDIndicator = request.Body<const Config>()->FPDIndicator.Value();
-                    uint32_t FPDColor = request.Body<const Config>()->FPDColor.Value();
-                    bool ToPersist = request.Body<const Config>()->ToPersist.Value();
-                    if (_controller->SetFPDColor(FPDIndicator, FPDColor, ToPersist)) {
+                } else if (index.Current() == _T("FPDColor") && (request.HasBody())) {
```
Comment (pwielders) [inline FrontPanel/FrontPanel.cpp]: Sama as above, DColor. remove the FP.

File: FrontPanel/FrontPanel.h [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+    private:
+        Core::ProxyType<Web::Response> GetMethod(Core::TextSegmentIterator& index);
+        Core::ProxyType<Web::Response> PostMethod(Core::TextSegmentIterator& index, const Web::Request& request);
+        Core::ProxyType<Web::Response> DeleteMethod(Core::TextSegmentIterator& index);
+
+    private:
+        uint8_t _skipURL;
+        PluginHost::IShell* _service;
+        Core::ProxyType<WPASupplicant::FrontPanelHAL> _controller;
```
[the remark this answers is not in the document]
Comment (aswaunni) [reply FrontPanel/FrontPanel.h]: I have made these changes in the latest commit.

**Pull request merged**